### ISSUE TITLE
How to add custom JSON converters?

### ISSUE THREAD

**Issue by RobTF** [OData/AspNetCoreOData#222]
Hi,

I see there is a new package for adding Netwonsoft JSON so I'm guessing there are available extensibility points for controlling the JSON serialization.

We are using System.Text.Json, but would like to customise how certain entities are serialized/deserialized. The OData library appears to ignore the normal ASP.NET Core MVC JSON settings, so is there some other way of doing this?

many thanks,

**Comment by xuzhg** [comment]
@RobTF You are right. ODL uses its own JSON writer/reader.  ODL own writer/reader was designed and implemented before the Newtonsoft.JSON and System.Text.JSON.  
Even though, ODL provides the [IJsonWriterFactory](https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/Json/IJsonWriterFactory.cs) and [IJsonReaderFactory](https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Core/Json/IJsonReaderFactory.cs) that you can customize.

**Comment by RobTF** [comment]
Hi Sam,

Thanks for the details - those interfaces look very low level though. Do you have any examples of an implementation whereby a particular data type is serialized/deserialized in a different way?

To be specific, we are working with the Geometry types from the NettopologySuite assemblies, as this is what EF Core works with, but are having issues getting these out of the ODATA endpoints.

**Comment by RobTF** [comment]
Hi, I just noticed that 8.0.1 of the library is now released and exposes some new interfaces such as `ODataSerializerProvider`. Will this version be able to help with the issue above?

**Comment by TehWardy** [comment]
Maybe create your own formatter ?
I built my own for CSV, XML and Excel dumps from OData.
I don't see why this wouldn't work for JSON too.

**Comment by chainavawongse** [comment]
Hi @RobTF, did you find the solution? IJsonWriter is quite a handful to implement.
